api/src/main/java/org/openmrs/module/eptsreports/reporting/library/queries/TXCurrQueries.java

```diff
@@ -471,120 +471,81 @@ public static String getPatientWithoutScheduledDrugPickupDateMasterCardAmdArtPic
-            + " ON         pa.patient_id=en.patient_id "
-            + " INNER JOIN obs ob "
-            + " ON         en.encounter_id=ob.encounter_id "
-            + " WHERE      pa.voided=0 "
-            + " AND        en.voided=0 "
-            + " AND        ob.voided=0 "
-            + " AND        en.location_id=:location "
-            + " AND        ob.location_id=:location "
-            + " AND        en.encounter_type IN(${masterCardDrugPickupEncounterType}) "
-            + " AND        ob.concept_id     IN(${artDatePickup}) "
-            + " AND        ob.value_datetime IS NOT NULL "
-            + "AND        ob.value_datetime<=:onOrBefore ) fn) ";
+            + "FROM   patient pat "
+            + "WHERE  pat.voided=0 "
+            + "    AND    pat.patient_id NOT IN( "
+            + "        SELECT fn.patient_id "
+            + "        FROM  ( "
+            + "            SELECT     qa.patient_id "
+            + "            FROM ( "
+            + "                SELECT pat.patient_id,Max(e.encounter_datetime) AS encounter_datetime "
+            + "                FROM  patient pat "
+            + "                    INNER JOIN encounter e ON  pat.patient_id=e.patient_id "
+            + "                WHERE      e.encounter_datetime<= :onOrBefore "
+            + "                    AND        pat.voided=0 "
+            + "                    AND        e.voided=0 "
+            + "                    AND        e.location_id= :location "
+            + "                    AND        e.encounter_type IN(${6},${9}) "
+            + "                GROUP BY   pat.patient_id "
+            + "                ) qa "
+            + "                INNER JOIN encounter e1 ON qa.patient_id=e1.patient_id "
+            + "                INNER JOIN obs o1 ON e1.encounter_id=o1.encounter_id "
+            + "            WHERE      qa.encounter_datetime=e1.encounter_datetime "
+            + "                AND        e1.encounter_datetime<= :onOrBefore "
+            + "                AND        e1.voided=0 "
+            + "                AND        e1.encounter_type IN(${6},${9}) "
+            + "                AND        e1.location_id= :location "
+            + "                AND        o1.value_datetime IS NOT NULL "
+            + "                AND        o1.voided=0 "
+            + "                AND        o1.concept_id IN(${1410}) "
+            + "                AND        o1.location_id = :location "
+            + "            UNION "
+            + "            SELECT     qc.patient_id "
+            + "            FROM ( "
+            + "                SELECT     pat.patient_id, max(e.encounter_datetime) AS encounter_datetime "
+            + "                FROM       patient pat "
+            + "                    INNER JOIN encounter e ON         pat.patient_id=e.patient_id "
+            + "                WHERE      e.encounter_datetime<= :onOrBefore "
```

Comment:
Put some space after the e.encounter_datetime before the <= only helps for readability

Reply:
done